bukkit/rpk-unconsciousness-bukkit/src/main/kotlin/com/rpkit/unconsciousness/bukkit/listener/EntityDamageByEntityListener.kt

```diff
@@ -38,19 +38,19 @@ class EntityDamageByEntityListener(private val plugin: RPKUnconsciousnessBukkit)
             val minecraftProfileService = Services[RPKMinecraftProfileService::class.java] ?: return
             val characterService = Services[RPKCharacterService::class.java] ?: return
             val unconsciousnessService = Services[RPKUnconsciousnessService::class.java] ?: return
-            val minecraftProfile = minecraftProfileService.getPreloadedMinecraftProfile(damager) ?: return
-            val character = characterService.getPreloadedActiveCharacter(minecraftProfile) ?: return
-            if (!unconsciousnessService.isUnconscious(character)) return
+            val minecraftProfile = minecraftProfileService.getMinecraftProfile(damager) ?: return
+            val character = characterService.getActiveCharacter(minecraftProfile) ?: return
+            if (!unconsciousnessService.isUnconscious(character).join()) return
```

Comment:
You have a CompletableFuture instead of a character being passed here, maybe use getPreloadedMinecraftProfile & getPreloadedActiveCharacter as the player will always be online when they are being damaged so they will be loaded.
Also EntityDamageByEntityEvent is fired synchronously so you shouldn't join here - perhaps preloading unconscious state in the unconsciousness service is required to enable the event to be cancelled synchronously.